src/megaclient.cpp

```diff
@@ -1340,9 +1375,31 @@ void MegaClient::exec()
                                 fetchnodes(true);
                                 reqtag = creqtag;
                             }
+                            else if (e == API_EAGAIN || e == API_ERATELIMIT)
+                            {
+                                if (!statecurrent)
+                                {
+                                    fnstats.eagains++;
+                                }
+                            }
                         }
                         // fall through
                     case REQ_FAILURE:
+                        if (pendingsc && !statecurrent)
```

Comment:
We can use one single `if` block by adding the condition `&& pendingsc->httpstatus != 200` to this one